addon/components/sl-select.js

```diff
@@ -141,58 +209,83 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
-                    ) {
-                        return Ember.get( item, this.get( 'optionLabelPath' ) );
-                    }
+                }
 
-                    return item;
-                },
+                return item;
+            };
+        }
+    ),
 
-                id: ( item ) => {
-                    let value = item;
-                    const typeOfItem = Ember.typeOf( item );
+    /**
+     * Select2 fomartSelection function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2FormatSelection: Ember.computed(
+        function() {
+            return ( item ) => {
+                const typeOfItem = Ember.typeOf( item );
+
+                if (
+                    'object' === typeOfItem ||
+                    'instance' === typeOfItem
+                ) {
+                    return Ember.get( item, this.get( 'optionLabelPath' ) );
+                }
 
-                    if (
-                        'object' === typeOfItem ||
-                        'instance' === typeOfItem
-                    ) {
-                        const optionValuePath = this.get( 'optionValuePath' );
-                        value = Ember.get( item, optionValuePath );
-                    }
+                return item;
+            };
+        }
+    ),
 
-                    return value;
-                },
+    /**
+     * Select2 id function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2Id: Ember.computed(
+        function() {
+            return ( item ) => {
+                const typeOfItem = Ember.typeOf( item );
+
+                if (
+                    'object' === typeOfItem ||
+                    'instance' === typeOfItem
+                ) {
+                    return Ember.get( item, this.get( 'optionValuePath' ) );
+                }
 
-                initSelection: ( element, callback ) => {
-                    const value = element.val();
+                return item;
+            };
+        }
+    ),
 
-                    if ( !value || !value.length ) {
-                        return callback( [] );
-                    }
+    /**
+     * Select2 initSelection function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2InitSelection: Ember.computed(
+        function() {
+            return ( element, callback ) => {
+                const value = element.val();
 
+                if ( !value || !value.length ) {
+                    callback( [] );
+                } else {
                     const content = this.get( 'content' );
                     const contentLength = content.length;
                     const filteredContent = [];
                     const multiple = this.get( 'multiple' );
                     const optionValuePath = this.get( 'optionValuePath' );
-                    const values = value.split( ',' );
+                    const values = multiple ? value.split( ',' ) : [ value ];
```

Comment:
My thought process went something like if the user wanted an option with a comma in it. It wouldn't make sense if a user had a single select and passed value="123,123", and actually had an option "123,123". By splitting on the comma the selection would end up disabled since it can't find 123 in the options.